src/algebraic_geometry/projective_spectrum/scheme.lean

```diff
@@ -290,34 +226,29 @@ def to_fun (x : Proj.T| (pbo f)) : (Spec.T (A⁰_ f_deg)) :=
   rcases x.1.is_prime.mem_or_mem (show a1 * a2 * f ^ N * f ^ M ∈ _, from _) with h1|rid2,
   rcases x.1.is_prime.mem_or_mem h1 with h1|rid1,
   rcases x.1.is_prime.mem_or_mem h1 with h1|h2,
-  { left,
-    rw mem_carrier_iff,
-    simp only [show (mk a1 ⟨f ^ n1, _⟩ : away f) = mk a1 1 * mk 1 ⟨f^n1, ⟨n1, rfl⟩⟩,
+  { left, simp only [show (mk a1 ⟨f ^ n1, _⟩ : away f) = mk a1 1 * mk 1 ⟨f^n1, ⟨n1, rfl⟩⟩,
       by rw [localization.mk_mul, mul_one, one_mul]],
     exact ideal.mul_mem_right _ _ (ideal.subset_span ⟨_, h1, rfl⟩), },
-  { right,
-    rw mem_carrier_iff,
-    simp only [show (mk a2 ⟨f ^ n2, _⟩ : away f) = mk a2 1 * mk 1 ⟨f^n2, ⟨n2, rfl⟩⟩,
+  { right, simp only [show (mk a2 ⟨f ^ n2, _⟩ : away f) = mk a2 1 * mk 1 ⟨f^n2, ⟨n2, rfl⟩⟩,
       by rw [localization.mk_mul, mul_one, one_mul]],
     exact ideal.mul_mem_right _ _ (ideal.subset_span ⟨_, h2, rfl⟩), },
   { exact false.elim (x.2 (x.1.is_prime.mem_of_pow_mem N rid1)), },
   { exact false.elim (x.2 (x.1.is_prime.mem_of_pow_mem M rid2)), },
   { rw [mul_comm _ (f^N), eq1],
     refine mul_mem_right _ _ (mul_mem_right _ _ (sum_mem _ (λ i hi, mul_mem_left _ _ _))),
-    generalize_proofs h,
-    exact (classical.some_spec h).1 },
+    generalize_proofs h, exact (classical.some_spec h).1 },
 end⟩
 
 /-
 The preimage of basic open set `D(a/f^n)` in `Spec A⁰_f` under the forward map from `Proj A` to
 `Spec A⁰_f` is the basic open set `D(a) ∩ D(f)` in  `Proj A`. This lemma is used to prove that the
 forward map is continuous.
 -/
-lemma preimage_eq (a : A) (n : ℕ)
-  (a_mem_degree_zero : (mk a ⟨f ^ n, ⟨n, rfl⟩⟩ : away f) ∈ A⁰_ f_deg) :
-  to_fun 𝒜 f_deg ⁻¹'
-      ((sbo (⟨mk a ⟨f ^ n, ⟨_, rfl⟩⟩, a_mem_degree_zero⟩ : A⁰_ f_deg)) :
-        set (prime_spectrum {x // x ∈ A⁰_ f_deg}))
+lemma preimage_eq' (a b : A) (k : ℕ) (a_mem : a ∈ 𝒜 k) (b_mem1 : b ∈ 𝒜 k)
```

Comment:
Is the prime in the name necessary? I did not find a non-primed version.

Reply:
you are right! I have removed this in [ee3ae68](https://github.com/leanprover-community/mathlib/pull/16693/commits/ee3ae68a5e581e0469fa0aa5c34e593986ef4b78)